src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioProjectFactory.cs

```diff
@@ -66,7 +67,8 @@ public VisualStudioProject CreateAndAddToWorkspace(string projectSystemName, str
                         language: language,
                         filePath: creationInfo.FilePath,
                         compilationOptions: creationInfo.CompilationOptions,
-                        parseOptions: creationInfo.ParseOptions);
+                        parseOptions: creationInfo.ParseOptions)
+                    .WithTelemetryId(creationInfo.ProjectGuid);
```

Comment:
We're prepared for all the potential wrongness of this? (Duplicate projects with the same GUID, the GUID being generated at random sometimes, etc.)

Reply:
I think the caveats are known. The value comes from the combination of SessionId with the ProjectId (sessionid is always unique), not with the projectId over time.